src/main/java/gregtech/api/capability/impl/AbstractRecipeLogic.java

```diff
@@ -542,7 +567,13 @@ protected void setupRecipe(Recipe recipe) {
         setMaxProgress(resultOverclock[1]);
         this.recipeEUt = resultOverclock[0];
         this.fluidOutputs = GTUtility.copyFluidList(recipe.getFluidOutputs());
-        this.itemOutputs = GTUtility.copyStackList(recipe.getResultItemOutputs(getOutputInventory().getSlots(), GTUtility.getTierByVoltage(recipeEUt)));
+        if (canDoChancedOutputs()) {
+            this.itemOutputs = GTUtility.copyStackList(recipe.getResultItemOutputs(getOutputInventory().getSlots(), GTUtility.getTierByVoltage(recipeEUt)));
+        } else {
+            this.itemOutputs = GTUtility.copyStackList((recipe.getOutputs()).subList(0, Math.min(recipe.getOutputs().size(), getOutputInventory().getSlots())));
+        } if (clampOutputs()) {
+            this.itemOutputs = GTUtility.copyStackList(itemOutputs.subList(0, 1));
+        }
```

Comment:
Can we short-circuit this a bit more? Lots of copying ItemStacks in gametime logic isn't the best thing to do